[Bug]: bt-tether.py plugin not working on different locales/language

### What happened?

When pairing with the smartphone using `bluetoothctl`, it creates a network called "MySmartphone Network" by default.

Therefore, the `bt-tether.py` plugin assumes this is the network name, [doing the following](https://github.com/jayofelony/pwnagotchi/blob/noai/pwnagotchi/plugins/default/bt-tether.py#L181):
`self.phone_name = self.options["phone-name"] + " Network"`

However, the network name generated varies depending on the locales/language. E.g. "Red MySmartphone" (Spanish). Therefore, the plugin cannot find the network and the connection fails ([here](https://github.com/jayofelony/pwnagotchi/blob/noai/pwnagotchi/plugins/default/bt-tether.py#L194)).

It would be good to handle these cases. Perhaps with something like:
```
self.phone_name = [
    name.strip()
    for name in self.nmcli(["-f", "NAME", "connection", "show"]).stdout.splitlines()
    if self.phone_name in name
][0]
```

A workaround is to change the network name manually using `nmcli`:
`nmcli connection modify "Red MySmartphone" connection.id "MySmartphone Network"`



### Version

2.9.5.3

### 3rd Party Hardware

Yes

### Relevant log output

```shell
21:25:35 [ERROR] [Thread-4] : [BT-Tether] Failed to connect to device: Command '['nmcli', 'connection', 'modify', 'MySmartphone Network', 'connection.type', 'bluetooth', 'bluetooth.type', 'panu', 'bluetooth.bdaddr', '<MY_MAC>', 'ipv4.method', 'manual', 'ipv4.dns', '8.8.8.8 1.1.1.1', 'ipv4.addresses', '192.168.44.2/24', 'ipv4.gateway', '192.168.44.1', 'ipv4.route-metric', '100']' returned non-zero exit status 10.
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/pwnagotchi/plugins/default/bt-tether.py
+++ b/pwnagotchi/plugins/default/bt-tether.py
@@ -154,6 +154,28 @@ class BTTether(plugins.Plugin):
     def nmcli(self, args, pattern=None):
         return self.exec_cmd("nmcli", args, pattern)
 
+    def _get_bt_connection_name(self):
+        # Get bluetooth connections that contains the phone-name
+        bt_connections = [
+            connection.split(":")[0].strip()
+            for connection
+            in self.nmcli(["-t", "-f", "NAME,TYPE", "connection", "show"]).stdout.splitlines()
+            if connection.endswith(":bluetooth") and self.options["phone-name"] in connection
+        ]
+
+        if len(bt_connections) == 1:
+            return bt_connections[0]
+        
+        if len(bt_connections) > 1:
+            # Multiple matches found: check the MAC
+            for bt_connection in bt_connections:
+                if self.nmcli(["-f", "bluetooth.bdaddr", "connection", "show", bt_connection], self.options["mac"].upper()) != -1:
+                    return bt_connection
+
+        # Fallback to the old logic
+        logging.error(f"[BT-Tether] Connection not found for {self.options['phone-name']}git")
+        return self.options["phone-name"] + " Network"
+
     def on_loaded(self):
         logging.info("[BT-Tether] plugin loaded.")
 
@@ -178,7 +200,7 @@ class BTTether(plugins.Plugin):
             logging.error(f"[BT-Tether] IP error: {address}")
             return
 
-        self.phone_name = self.options["phone-name"] + " Network"
+        self.phone_name = self._get_bt_connection_name()
         self.mac = self.options["mac"].upper()
         dns = self.options.get("dns", "8.8.8.8 1.1.1.1")
         if not re.match(DNS_PTTRN, dns):

--- a/pwnagotchi/plugins/default/bt-tether.py
+++ b/pwnagotchi/plugins/default/bt-tether.py
@@ -179,7 +179,7 @@ class BTTether(plugins.Plugin):
             return
 
         self.phone_name = self.options["phone-name"] + " Network"
-        self.mac = self.options["mac"]
+        self.mac = self.options["mac"].upper()
         dns = self.options.get("dns", "8.8.8.8 1.1.1.1")
         if not re.match(DNS_PTTRN, dns):
             if dns == "":

